common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx

```diff
@@ -56,24 +61,52 @@ export default class ConfirmationModal extends React.Component {
       }
     }, 1000);
 
-    this.props.wallet.getAddress().then(address => {
-      this.setState({ address });
-    });
+    this._setWalletAddress(this.props.wallet);
   }
 
   componentWillUnmount() {
     clearTimeout(this.readTimer);
   }
 
+  _setWalletAddress(wallet: BaseWallet) {
+    wallet.getAddress().then(fromAddress => {
```

Comment:
No ideas who did this, but this would complicate a lot of our code. If hw wallets need async address resolution they should do it on unlock cc @skubakdj